src/main/java/BasketInformations.java

```diff
@@ -0,0 +1,98 @@
+package com.company;
+
+import java.util.HashMap;
+
+/**
+ * This class is a basket
+ */
+public class BasketInformations {
+
+	// The product of the basket
+	static HashMap<String, Integer> map = new HashMap<String, Integer>();
+
+	static String[] localProducts = new String[]{"123", "222", "44", "657"};
+
+	// The fact that the basket has promo code
+	private static boolean codeDePromotion = false;
+
+	public void addProductToBasket(String product, Integer price, boolean isPromoCode) {
+		if (isPromoCode) {
```

Comment:
I downloaded the code and just realizing that the there is no data structure representing the product.

Instead of passing it around, perhaps you could add a model class that represents a product:
```java
public class BasketProduct {
   private String productId;
   private double price; 
}```

Then your cache definition would be come:
```
private final HashMap<String, Product> map = new HashMap<>();
```